parser_library/src/workspaces/file_manager_impl.cpp

```diff
@@ -409,32 +418,32 @@ utils::resource::resource_location file_manager_impl::get_virtual_file_workspace
     return utils::resource::resource_location();
 }
 
-open_file_result file_manager_impl::update_file(const utils::resource::resource_location& document_loc)
+utils::value_task<open_file_result> file_manager_impl::update_file(
+    const utils::resource::resource_location& document_loc)
 {
-    std::unique_lock lock(files_mutex);
-
-    if (auto f = m_files.find(document_loc); f == m_files.end() || f->second.file->get_lsp_editing())
-        return open_file_result::identical;
-
-    lock.unlock();
-
-    auto current_text = m_file_reader->load_text(document_loc);
+    {
+        std::lock_guard lock(files_mutex);
 
-    lock.lock();
+        if (auto f = m_files.find(document_loc); f == m_files.end() || f->second.file->get_lsp_editing())
+            return {};
```

Comment:
Ok, so this is something I struggled with a bit...
The idea here (and other places) is that the return value here is an empty task - as in "nothing to do". This is something that will need to be addressed in general, how to deal with these cases where we might need to allocate a coroutine frame in general, but in most cases we have nothing to do and it would be just a waste of resources.